---
bibliography: /home/beau/repos/life/dox/acad.bib
title: "Geno-Clinical Model for the Diagnosis of Bone Marrow Myeloid Neoplasms"
author:
  - name: C. Beau Hilton, BA
    institute: [1,2]
    email: hiltonc@ccf.org
  - name: Manja Meggendorfer, PhD
    institute: [3]
    email: manja.meggendorfer@mll.com
  - name: Mikkael A. Sekeres, MD, MS
    institute: [1]
    email: sekerem@ccf.org
  - name: Jacob Shreve, MD, MS
    institute: [4]
    email: shrevej2@ccf.org
  - name: Nathan Radakovich
    institute: [2]
    email: radakon@ccf.org
  - name: Yazan Rouphail
    institute: [5]
    email: rouphail.2@buckeyemail.osu.edu
  - name: Wencke Walter, PhD
    institute: [3]
    email: wencke.walter@mll.com
  - name: Stephan Hutter, PhD
    institute: [3]
    email: stephan.hutter@mll.com
  - name: Eric Padron, MD
    institute: [9]
    email: eric.padron@moffitt.org
  - name: Michael R. Savona, MD
    institute: [6, 10]
    email: michael.savona@vanderbilt.edu
  - name: Aaron T. Gerds, MD, MS
    institute: [1]
    email: gerdsa@ccf.org
  - name: Sudipto Mukherjee, MD, PhD, MPH
    institute: [1]
    email: mukhers2@ccf.org
  - name: Brianna N Smith, MD, MS
    institute: [6, 7, 10]
    email: brianna.n.smith@vumc.org
  - name: Yasunobu Nagata, MD, PhD
    institute: [8]
    email: nagatay@ccf.org
  - name: Cassandra M. Hirsch
    institute: [8]
    email: hirschc@ccf.org
  - name: Rami S. Komrokji, MD
    institute: [9]
    email: rami.komrokji@moffitt.org
  - name: Babal K. Jha, PhD
    institute: [8]
    email: jhab@ccf.org
  - name: Claudia Haferlach, MD
    institute: [3]
    email: claudia.haferlach@mll.com
  - name: Jaroslaw P. Maciejewski, MD, PhD
    institute: [8]
    email: maciejj@ccf.org
  - name: Torsten Haferlach, MD
    institute: [3]
    email: torsten.haferlach@mll.com
  - name: Aziz Nazha, MD
    institute: [1]
    email: nazhaa@ccf.org
    correspondence: true
institute:
  - 1: Leukemia Program, Department of Hematology and Medical Oncology, Taussig Cancer Institute, Cleveland Clinic, Cleveland, OH
  - 2: Cleveland Clinic Lerner College of Medicine of Case Western Reserve University, Cleveland, OH
  - 3: MLL Munich Leukemia Laboratory, Munich, Bavaria, Germany
  - 8: Department of Translational Hematology and Oncology Research, Taussig Cancer Institute, Cleveland Clinic, Cleveland, OH
  - 4: Department of Internal Medicine, Cleveland Clinic, Cleveland, OH
  - 5: Ohio State University, Columbus, OH
  - 6: Department of Medicine, Vanderbilt University Medical Center, Nashville, TN
  - 7: Department of Pediatrics, Vanderbilt University School of Medicine, Nashville, TN
  - 10: Program in Cancer Biology, Vanderbilt University Medical Center, Nashville, TN
  - 9: Department of Malignant Hematology, H. Lee Moffitt Cancer Center and Research Institute, Tampa, FL

date: "`r format (Sys.time(), format='%d %B, %Y')`"
header-includes:
    - \usepackage{setspace}\doublespacing
indent: true
output:
  word_document:
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
      - '--lua-filter=author-info-blocks.lua'
  pdf_document:
    latex_engine: xelatex
    pandoc_args:
      - '--lua-filter=scholarly-metadata.lua'
      - '--lua-filter=author-info-blocks.lua'
---

```{r, echo=FALSE}
library(reticulate)
use_condaenv("lgb_shap")
```

\tableofcontents

# Abstract
## Background

Myelodysplastic syndromes (MDS) and other myeloid neoplasms are mainly diagnosed based on morphological changes in the bone marrow. Diagnosis can be challenging in patients (pts) with pancytopenia with minimal dysplasia, and is subject to interobserver variability, with up to 40% disagreement in diagnosis (Zhang, ASH 2018). Somatic mutations can be identified in all myeloid neoplasms, but no gene or set of genes are diagnostic for each disease phenotype. We developed a geno-clinical model that uses mutational data, peripheral blood values, and clinical variables to distinguish among several bone marrow disorders.

## Methods

We combined genomic and clinical data from 2471 pts treated at our institution (684) and the Munich Leukemia Laboratory (1787). Pts were diagnosed with MDS, idiopathic cytopenia of undetermined significance (ICUS), clonal cytopenia of undetermined significance (CCUS), chronic myelomonocytic leukemia (CMML), polycythemia vera (PV), essential thrombocythemia (ET), and myelofibrosis (PMF) according to 2016 WHO criteria. Diagnoses were confirmed by independent hematopathologists not associated with the study. A panel of 60 genes commonly mutated in myeloid malignancies was included. The cohort was randomly divided into learner (80%) and validation (20%) cohorts. Machine learning algorithms were applied to predict the phenotype. Prediction performance was evaluated according to the area under the curve of the receiver operator characteristic (ROC-AUC) and other metrics as appropriate. Feature extraction algorithms were used to produce interpretable differential diagnoses for each pt.

## Results

Of 2471 pts, 1306 had MDS, 223 had ICUS, 107 had CCUS, 478 had CMML, 89 had MDS/MPN, 79 had PV, 90 had ET, and 99 had PMF. The median age for the entire cohort was 71 years (range, 9-102); 38% were female. The median white blood cell count (WBC) was 3.2x10ˆ9/L (range, 0.00-179), absolute monocyte count (AMC) 0.21x10ˆ9/L (range, 0-96), absolute lymphocyte count (ALC) 0.88x10ˆ9/L (range, 0-357), absolute neutrophil count (ANC) 0.60x10ˆ9/L (range, 0-170), and hemoglobin (Hgb) 10.50 g/dL (range, 3.9-24.0). The most commonly mutated genes in all pts were: TET2 (28%), ASXL1 (23%), SF3B1 (15%).

Overall, the most important variables for prediction were: age, AMC, ANC, Hgb, Plt, ALC, total number of mutations, *JAK2*, *ASXL1*, *TET2*, *U2AF1*, *SRSF2*, *SF3B1*, *BCOR*, *EZH2*, and *DNMT3A*.

When applying the model to the validation cohort, AUC performance was at or above 0.95 for all diagnoses. When the analysis was restricted to MDS, ICUS, and CCUS, AUC remained high: 0.95. Finally, the model explanations can be induced to provide a reasoned, quantitative differential diagnosis. This is accomplished by showing a predicted probability and the composition of the prediction for each diagnosis under consideration for each pt.

## Conclusions

We propose a new approach using interpretable, individualized modeling to predict myeloid neoplasm phenotypes based on genomic and clinical data without bone marrow biopsy data. This approach can aid clinicians and hematopathologists when encountering pts with cytopenias and suspicion for these disorders. The model also provides feature attributions that allow for quantitative understanding of the complex interplay among genotypes, clinical variables, and phenotypes. Finally, this method can be used to produce quantitative differential diagnosis support for any set of diseases.

# Introduction

Myeloproliferative neoplasms (MPN), myelodysplastic syndromes (MDS), and the MDS/myeloproliferative neoplasm (MDS/MPN) overlap syndromes are myeloid neoplasms with varying prognoses and treatment strategies. Additionally, there is increasing recognition of the prevalence and importance of the related states of clonal hematopoiesis of indeterminate potential (CHIP, not covered in the present study), ICUS, and CCUS. CHIP describes the finding of acquired clonal mutations similar to those in MDS, but in apparently healthy individuals. ICUS describes single or multiple cytopenias without an explanation or known clonal mutations, even after BMBx evaluation. CCUS describes clonal mutations found with cytopenia(s) that do not meet criteria for a WHO-defined hematologic neoplasm (Bejar, 2016; Valent, 2019). Diagnosis of myeloid neoplasms and these related states is often challenging, and most require BMBx for morphologic and karyotypic analysis (Arber et al., 2016). The difficulty is magnified in pts without the expected karyotypic or morphologic changes on BMBx, especially in pts with unsuccessful biopsies (for example, due to marked hypercellularity or extensive myelofibrosis) (Humphries, 1990). Additionally, morphologic diagnosis is subject to interobserver variability, with up to 40% disagreement in diagnosis (Zhang et al, ASH 2018***; Arber et al., 2016; Valent et al., 2017). Genome sequencing technologies have identified recurrent somatic and germline mutations that play a role in pathophysiology, progression, and response to therapy. These may assist presumptive diagnoses in the absence of morphologic or karyotypic confirmation. However, with few exceptions these are not considered sufficiently specific for final diagnosis, particularly in the case of ICUS, which is a diagnosis of exclusion (Arber et al., 2016; Haferlach et al., 2014; Valent et al., 2017; Xu et al., 2017).

Machine learning (ML) is a family of computational algorithms that extract information from data by learning from relationships, patterns, and trends (Obermeyer & Emanuel, 2016). ML can produce powerful, reliable, and reproducible predictive models based on large and complex datasets (Kourou, Exarchos, Exarchos, Karamouzis, & Fotiadis, 2015). As the number of genomic and clinical variables under consideration by researchers and clinicians grows, increasingly sophisticated models will be needed to account for these variables and their interactions. Though ML models were traditionally “black boxes,” with high performance but low interpretability, modern methods make it possible to explain and therefore audit even the most complex algorithms, and this may in turn generate new avenues of inquiry (Lundberg and Lee, 2017; Zhao and Hastie, 2017). Additionally, clinician-built ML models may be designed to align more closely with the practice of medicine, particularly in their selection of outcome metrics and reporting tools, in order to augment decision-making, rather than replace it (Sniderman 2015).

# Methods

We combined genomic and clinical data from 2471 pts treated at our institution (684) and the Munich Leukemia Laboratory (1787). Pts were diagnosed with MDS, ICUS, CCUS, CMML, MDS/MPN, PV, ET, and PMF according to 2016 WHO criteria. Diagnoses were confirmed by independent hematopathologists not associated with the study. A panel of 60 genes commonly mutated in myeloid malignancies was included.

Key characteristics of the cohort were described using visualizations and descriptive statistics. Extensive exploratory visualizations are available in the Supplemental Appendix.

For modeling the cohort was randomly divided into either learner (80%) and validation (20%) cohorts, or split iteratively into similarly sized learner and validation cohorts to obtain cross-validated results. GBM algorithms were applied to predict the phenotype. SHAP was used to extract genomic/clinical variables that impacted the prediction, to visualize the impact of each variable on phenotype, and to generate individual predictions with explanations. Prediction performance was evaluated according to the BSL, AUC, confusion matrices, and inspection of individual predictions.

The code used to generate this analysis and manuscript is available upon request.

All analysis was done in Python and R, using data analysis and ML packages such as scikit-learn (Pedregosa 2011), LightGBM, pandas, TableOne, matplotlib, and others <++> include R packages.

# Results and Conclusions

Of 2471 pts, 1306 had MDS, 223 had ICUS, 107 had CCUS, 478 had CMML, 89 had MDS/MPN, 79 had PV, 90 had ET, and 99 had PMF. The median age for the entire cohort was 71 years (range, 9-102); 38% were female. The median white blood cell count (WBC) was 3.2x10ˆ9/L (range, 0.00-179), absolute monocyte count (AMC) 0.21x10ˆ9/L (range, 0-96), absolute lymphocyte count (ALC) 0.88x10ˆ9/L (range, 0-357), absolute neutrophil count (ANC) 0.60x10ˆ9/L (range, 0-170), and hemoglobin (Hgb) 10.50 g/dL (range, 3.9-24.0). (Table 1***).

The most commonly mutated genes in all pts were: TET2 (28%), ASXL1 (23%), SF3B1 (15%). In MDS, they were: TET2 (26%), SF3B1 (24%), ASXL1 (21%). In CCUS: TET2 (46%), SRSF2 (24%), ASXL1 (23%). In CMML, TET2 (51%), ASXL1 (43 %), SRSF2 (25%). In MDS/MPN: SF3B1 (39%), JAK2 (37%), TET2 (20%). In PV, JAK2 (94%), TET2 (22%), DNMT3A (8%). In ET: JAK2 (44%), TET2 (13%), DNMT3A (8%). In PMF: JAK2 (67%), ASXL1 (43%), SRSF2 (17%) (Table 1***).

When applying the model to the validation cohort, AUC performance was as follows (a perfect predictor has an AUC of 1, and AUC ≥ 0.90 are generally considered excellent): MDS: 0.95 ± 0.04, ICUS: 0.96 ± 0.05, CCUS: 0.95 ± 0.05, CMML: 0.95 ± 0.05, MDS/MPN: 0.95 ± 0.05, PV: 0.95 ± 0.05, ET: 0.96 ± 0.05, PMF: 0.95 ± 0.05. When the analysis was restricted to MDS, ICUS, and CCUS, the AUC remained high, 0.95 ± 0.4. Precision-Recall (PR) curves and the corresponding Average Precision (AP) value (analogous to AUC for ROC curves) were produced for each class in a cross-validation scheme, with APs as follows: MDS: 0.95 ± 0.04, ICUS: 0.96 ± 0.05, CCUS: 0.78 ± 0.23, CMML: 0.97 ± 0.03, MDS/MPN: 0.33 ± 0.12, PV: 0.93 ± 0.02, ET: 0.94 ± 0.03, PMF: 0.79 ± 0.12. The model was well calibrated. BSL were as follows: MDS: 0.08, ICUS: 0.02, CCUS: 0.02, CMML: 0.02, MDS/MPN: 0.03, PV: 0.01, ET: 0.01, PMF: 0.02.  See Table 2*** for a summary of performance.

A total of 71 genomic/clinical variables were evaluated. Feature extraction algorithms were used to identify the variables with the most significant impacts on prediction. The top variables are shown in Figure 1***. Overall, the most important variables were: AMC, ANC, Hgb, Plt, ALC, total number of mutations, AEC, ABC, WBC, JAK2, age, ASXL1, TET2, U2AF1, DNMT3A, SF3B1, female gender, SRSF2, EZH2, and STAG2. The most important variables for each class, along with the relative impact of high or low values for those variables, are shown in the Supplementary Appendix.

Quantitative differential diagnoses were produced for each pt in the validation cohort. A selection of these are included in the supplementary materials, and one is in Figure 2***. The first shows an accurate diagnosis, that is, the diagnosis with the highest probability matched the actual diagnosis. The second shows an inaccurate diagnosis. Features in blue decrease the probability of a given diagnosis, while features in red increase it. The size of the red or blue bar for a given feature reflects its relative contribution. The probability for each diagnosis is given in the “output value,” as a number between zero and one.

# Discussion

Interpretable, personalized quantitative differential diagnoses for pts with myeloid neoplasms were produced with a high degree of reliability using clinical and genetic data, exclusive of BMBx. The components of prediction were shown for the cohort overall, and two personalized predictions were shown. Clinical data outweigh mutational data for most predictions.

The model found the diagnostic criteria used in the literature, overall, such as JAK2 mutations for PV and ET, and lack of known mutations for ICUS. It was also generally successful at weighing the contributions of genetic and clinical data for individual predictions. However, as shown in the Supplementary Appendix, the model’s predictions reflect the most likely diagnosis based on the majority of samples within the database, and may therefore predict incorrectly when a sample does not fit the general trend. In this example, the degrees and natures of cytopenia were outweighed by the lack of a known mutation, whereas in others, details regarding cytopenias may outweigh the lack of mutation. As a diagnostic support, the prediction made by the model in this case is favorable though incorrect, as ICUS is a diagnosis of exclusion and it would be reasonable to pursue further diagnostic workup for MDS before deciding on treatment. This detail also speaks to the importance of the clinical scenario and domain knowledge in applying an ML tool, and the importance of being able to audit the components of prediction as well as the full set of possibilities considered by the model.

Our initial pilot project was restricted to differentiating MDS and CMML (Hilton et al 2018). It is significant that overall performance did not notably decrease from the two-class case (MDS vs. CMML) to the multi-class case (the current manuscript). There may be several factors contributing to this, such as the strength of clinical and genomic signal in diagnosing PV and ET, as well as the strong signal of lack of mutations for ICUS. In other words, certain of these diagnoses may be easy for the model to find, because they have more uniform and extreme phenotypes and genotypes that are easily exposed to the current model. It is also important to note that the performance from test to test varies widely for the diagnoses that are underrepresented in the dataset, as shown in the cross-validated PR curves.

The most important limitations of the findings under consideration are the limitations inherent in the diagnosis of myeloid neoplasms, the unknown generalizability of this model considering the size of the cohort and possible biases in the dataset, and the lack of outcome data for any treatments selected based on the predicted diagnoses.

Given, as above, that myeloid neoplasms are often difficult to diagnose, and that there is high heterogeneity in the interpretation of the gold standard BMBx even among highly trained hematopathologists, the ground truth upon which this analysis was built is not certain. Myeloid neoplasms also evolve over time, collecting genotypes and phenotypes. Certain diagnoses, such as CMML, require persistence of phenotype (in that case, monocytosis). The single time point considered for each pt in this cohort does not consider any changes or persistences. Despite representing many major categories of myeloid neoplasms, the diagnoses included in the model did not include other possible causes of cytosis or cytopenia of any cell line, such as infection or non-myeloid neoplasms. Therefore, although the model may perform well in the situation that a clinician has a high suspicion for one of the included diagnoses, the entire clinical picture must be taken into account. However, by presenting a probability in the context of every diagnosis under consideration, rather than a single diagnosis without any metric suggesting the degree of confidence, the risk of misuse is reduced.

The cohort included in this study is large and rich by the research standards in hematologic malignancy, but small or modest in sample size and narrow in features by machine learning standards. Useful tools such as calibration curves are not robustly defined for this small data. Additionally, the generalizability of prediction, particularly for classes with few examples, cannot be assured without increasing cohort size, preferably with a dataset from a third institution. A training-validation split of 80 and 20%, as well as cross-validation, help to reduce the uncertainty due to the small sample size, or at least report on it more fully, but cannot substitute for a larger sample size.

A group from Tel Aviv recently presented a similar project that used a GBM on clinical variables, without BMBx information, to predict MDS vs. not-MDS with AUC of 0.97, sensitivity of 88%, and specificity of 95% (Oster et al., 2018). This study used variable importance measures extracted directly from the GBM, which are not as reliable as those reported by Shapley explanations both theoretically and empirically, as they violate the principle of consistency required for reliable feature attribution, and in our experience are often different in size and magnitude to those found by the Shapley method. Additionally, the conclusion that many pts with MDS may not require BMBx may be premature, at least without rigorous examination of the dataset and the details of its production. Lastly, though more useful performance metrics may not have been reported for the same reasons we do not typically report them in summary format, there was not an indication of the reliability of probabilities of the model.

Shapley value estimations have not, to our knowledge, been used in any other formal academic study of diagnostic tools. They have been used successfully by the creators of the SHAP package in a recent paper reporting on a method to estimate the risk of hypoxia during surgery, in a real-time model that integrates a variety of features and outputs to a monitor available to the anesthetist. The time component of this study could be applied to the prediction of myeloid neoplasms if data were to be collected serially, and would help overcome the limitation described above. This study also appropriately reports probabilities, which is the only intuitive way to track risk in the minute-to-minute changes of an operating theater (Lundberg 2018).


@Agniel2018
* sdlije
* lkklje
* lkje

```{python}
# import lightgbm as lgb
# import shap
names = ["john", "lucy", "george"]
for name in names:
	print(name)
	print(name.upper())

```


# Figure Legends
## Figure 1
## Figure 2
## Figure 3

# Tables
## Table 1
## Table 2

# References


