\section{Abstract}%\subsection{Background}%

Myelodysplastic syndromes (MDS) and other myeloid neoplasms are mainly diagnosed based on morphological changes in the bone marrow. The diagnosis can be challenging in patients (pts) with pancytopenia with minimal dysplasia, and is subject to inter-observer variability. Somatic mutations can be identified in all myeloid neoplasms but no gene or set of genes are diagnostic for each disease phenotype. 
We developed a geno-clinical model that uses mutational data, peripheral blood values, and clinical variables to distinguish between several bone marrow disorders that include: MDS, idiopathic cytopenia of indeterminate significance (ICUS), clonal cytopenia of indeterminate significance (CCUS), MDS/MPN overlaps including chronic myelomonocytic leukemia (CMML), and myeloproliferative neoplasms such as polycythemia vera (PV), essential thrombocythemia (ET), and myelofibrosis (PMF).


\subsection{Methods}%

We combined genomic and clinical data from 2471 pts treated at our institution (684) and the Munich Leukemia Laboratory (1787). Pts were diagnosed with CCUS, CMML, ET, ICUS, MDS, MDS/MPN, PMF and PV according to 2008 WHO criteria. Diagnosis was confirmed by independent hematopathologists not associated with the study. A genomic panel of 22 genes commonly mutated in myeloid malignancies was included. The initial cohort was randomly (computer generated) divided into learner (80\%) and validation (20\%) cohorts. Multiple machine learning algorithms were applied to predict the phenotype. Feature extraction algorithms were used to extract genomic/clinical variables that impacted the algorithm decision and to visualize the impact of each variable on phenotype. Prediction performance was evaluated according to the area under the curve of the receiver operator characteristic (ROC-AUC) and confusion/accuracy matrices.


\subsection{Results}%

Of 2471 pts included, 1306 had MDS, 478 had CMML, 223 had ICUS, 107 had CCUS, 99 had PMF, 90 had ET, 89 had MDS/MPN, and 79 had PV. The median age for the entire cohort was 71 years (range, 9-102). 38\% were female. The median white blood cell count (WBC) was 3.2x10\textsuperscript{9}/L (range, 0.00-179), absolute monocyte count (AMC) 0.21x10\textsuperscript{9}/L (range, 0-96), absolute lymphocyte count (ALC) 0.88x10\textsuperscript{9}/L (range, 0-357), absolute neutrophil count (ANC) 0.60x10\textsuperscript{9}/L (range, 0-170), and hemoglobin (Hgb) 10.50 (range, 3.9-24.0). 

The most commonly mutated genes in all pts were: TET2 (28.0\%), ASXL1 (22.9\%), SF3B1 (15.2\%). In CCUS, they were: TET2 (45.8\%), SRSF2 (24.3\%), ASXL1 (23.4\%). In CMML, they were: TET2 (51.0\%), ASXL1 (42.9\%), SRSF2 (24.7\%). In ET, they were: JAK2 (44.4\%), TET2 (13.3\%), DNMT3A (7.8\%). In ICUS, they were: ASXL1 (0.0\%), BCOR (0.0\%), CBL (0.0\%). In MDS, they were: TET2 (26.1\%), SF3B1 (23.6\%), ASXL1 (20.8\%). In MDS/MPN, they were: SF3B1 (39.3\%), JAK2 (37.1\%), TET2 (20.2\%). In PMF, they were: JAK2 (66.7\%), ASXL1 (43.4\%), SRSF2 (17.2\%). In PV, they were: JAK2 (93.7\%), TET2 (21.5\%), DNMT3A (7.6\%).

The median total number of mutations/sample was 1 (range 0-8) for all pts, 1 (range 1-4) for CCUS, 2 (range 0-6) for CMML, 1 (range 0-4) for ET, 0 (range 0-0) for ICUS, 1 (range 0-8) for MDS, 2 (range 0-8) for MDS/MPN, 2 (range 0-5) for PMF and 1 (range 0-3) for PV.


A set of 55 genomic/clinical variables were evaluated and several feature extraction algorithms were used to identify the variables that have the most significant impact on the algorithm's decision. These variables included: (CCUS) ABC, ANC, Number of mutations, Plt, AMC, AEC, JAK2, Hgb, Age, DNMT3A, WBC, TET2, U2AF1, ALC, ASXL1, BCOR VUS, CBL, CBL VUS, BCOR, DNMT3A VUS; (CMML) AMC, AEC, WBC, ABC, ANC, ALC, Plt, Hgb, Age, TET2, Number of mutations, ASXL1, JAK2, SF3B1, SRSF2, RUNX1, Female gender, BCOR, BCOR VUS, CBL; (ET) ALC, Plt, Hgb, WBC, Age, AMC, ABC, ANC, AEC, JAK2, Number of mutations, SRSF2, DNMT3A VUS, FLT3, EZH2 VUS, EZH2, ETV6 VUS, ETV6, BCOR, DNMT3A; (ICUS) Number of mutations, ANC, ABC, AEC, WBC, Hgb, ALC, AMC, Plt, Female gender, Age, JAK2, TET2, FLT3, EZH2 VUS, ETV6, EZH2, ETV6 VUS, BCOR, DNMT3A VUS; (MDS) ANC, Plt, AMC, Hgb, ALC, WBC, ABC, JAK2, AEC, Age, Number of mutations, SRSF2, SF3B1, ASXL1, Female gender, TET2, FLT3 VUS, BCOR, BCOR VUS, CBL; (MDS/MPN) Hgb, ALC, Plt, AMC, JAK2, ANC, Age, WBC, AEC, SF3B1, ABC, Number of mutations, ASXL1, EZH2, Female gender, DNMT3A, BCOR, U2AF1, KRA, SRSF2; (PMF) JAK2, ABC, AMC, AEC, Hgb, ALC, ANC, Plt, Age, WBC, U2AF1, ASXL1, TET2, Female gender, Number of mutations, FLT3 VUS, BCOR, BCOR VUS, CBL, CBL VUS; (PV) Hgb, ALC, JAK2, WBC, Plt, AMC, AEC, ABC, Age, ANC, Number of mutations, ASXL1, BCOR, GATA2, CBL, CBL VUS, DNMT3A, DNMT3A VUS, ETV6, ETV6 VUS (Figure 1).

When applying the model to the validation cohort, performance was as follows: CCUS AUC: 0.97 +/- 0.04, precision: 0.87 +/- 0.30, recall: 0.68 +/- 0.30; CMML AUC: 0.99 +/- 0.01, precision: 0.89 +/- 0.02, recall: 0.91 +/- 0.02; ET AUC: 1.00 +/- 0.00, precision: 0.94 +/- 0.08, recall: 0.92 +/- 0.08; ICUS AUC: 0.99 +/- 0.01, precision: 0.87 +/- 0.08, recall: 0.76 +/- 0.08; MDS AUC: 0.97 +/- 0.01, precision: 0.87 +/- 0.05, recall: 0.95 +/- 0.05; MDS/MPN AUC: 0.86 +/- 0.04, precision: 0.70 +/- 0.26, recall: 0.26 +/- 0.26; PMF AUC: 0.96 +/- 0.03, precision: 0.76 +/- 0.09, recall: 0.56 +/- 0.09; PV AUC: 1.00 +/- 0.00, precision: 1.00 +/- 0.00, recall: 0.96 +/- 0.00; .

Individual pt data can also be entered into the model, with a probability of whether the diagnosis is MDS vs. CMML provided along with the impact of each variable on the decision, as shown in Figure 1.

When the analysis was restricted to mutations only, the accuracy of the model dropped dramatically (77\%, ROC-AUC .85).


\subsection{Conclusions}%

We propose a novel approach using interpretable, individualized modeling to predict myeloid neoplasm phenotypes based on genomic and clinical data without the need for bone marrow biopsy data. This approach can aid clinicians and hematopathologists when encountering pts with cytopenias and suspicion for these disorders. The model also provides feature attributions that allow for quantitative understanding of the complex interplay among genotypes, clinical variables, and phenotypes.



