\section{Results}%
Of 2202 pts included, 1391 pts had MDS, 531 pts had CMML, 107 pts had PMF, 91 pts had ET, and 82 pts had PV.The median age for the entire cohort was 72 years (range, 11-102). 38\% were female. The median white blood cell count (WBC) was 5.9x10^9/L (range, 0.60-144600), absolute monocyte count (AMC) 0.21x10^9/L (range, 0-96), absolute lymphocyte count (ALC) 0.83x10^9/L (range, 0-62), absolute neutrophil count (ANC) 2.81x10^9/L (range, 0-82840), and hemoglobin (Hgb)  10.40 (range, 3.9-24.0). 

The most commonly mutated genes in all pts were: JAK2 (8.4\%), ASXL1 (2.8\%), TET2 (2.0\%), DNMT3A (1.1\%), SRSF2 (1.1\%), U2AF1 (0.7\%), TP53 (0.4\%), SF3B1 (0.4\%), EZH2 (0.3\%), CBL (0.2\%), GATA2 (0.2\%), PRPF8 (0.2\%). In CMML, they were: ASXL1 (0.0\%), CBL (0.0\%), CEBPA (0.0\%), DNMT3A (0.0\%), ETV6 (0.0\%), EZH2 (0.0\%), GATA2 (0.0\%), IDH1 (0.0\%), IDH2 (0.0\%), JAK2 (0.0\%), KIT (0.0\%), KRAS (0.0\%). In ET, they were: JAK2 (44.0\%), TET2 (13.2\%), DNMT3A (7.7\%), ASXL1 (6.6\%), SRSF2 (3.3\%), SF3B1 (2.2\%), GATA2 (1.1\%), IDH1 (1.1\%), PRPF8 (1.1\%), TP53 (1.1\%), U2AF1 (1.1\%), CBL (0.0\%). In MDS, they were: ASXL1 (0.0\%), CBL (0.0\%), CEBPA (0.0\%), DNMT3A (0.0\%), ETV6 (0.0\%), EZH2 (0.0\%), GATA2 (0.0\%), IDH1 (0.0\%), IDH2 (0.0\%), JAK2 (0.0\%), KIT (0.0\%), KRAS (0.0\%). In PMF, they were: JAK2 (62.6\%), ASXL1 (46.7\%), SRSF2 (17.8\%), TET2 (15.0\%), U2AF1 (13.1\%), DNMT3A (10.3\%), TP53 (6.5\%), EZH2 (5.6\%), SF3B1 (5.6\%), CBL (4.7\%), GATA2 (2.8\%), NRAS (2.8\%). In PV, they were: JAK2 (93.9\%), TET2 (20.7\%), DNMT3A (7.3\%), ASXL1 (6.1\%), SRSF2 (2.4\%), IDH1 (1.2\%), TP53 (1.2\%), ZRSR2 (1.2\%), CBL (0.0\%), CEBPA (0.0\%), ETV6 (0.0\%), EZH2 (0.0\%).

The median total number of mutations/sample was 0.0 (range 0-6) for all pts, 0.0 (range 0-0) for CMML, 1.0 (range 0-4) for ET, 0.0 (range 0-0) for MDS, 2.0 (range 0-6) for PMF and 1.0 (range 0-3) for PV.


A set of 131 genomic/clinical variables were evaluated and several feature extraction algorithms were used to identify the variables that have the most significant impact on the algorithm's decision. These variables included: AMC, ALC, TET2, ANC, ASXL1, SF3B1, Hgb, number of mutations/sample, AEC, age, Plt, splenomegaly, RUNX1, NRAS, CBL, U2AF1, STAG2, DNMT3A, TP53, EZH2, SRSF2, and ZRSR2 (Figure 1).

When applying the model to the validation cohort, the ROC-AUC was .98 with an accuracy of 94\%, with other statistical values as follows: specificities CMML 93\%, MDS 96\%; sensitivities CMML 96\%, MDS 93\%; positive predictive values CMML 84\%, MDS 98\%; negative predictive values CMML 98\%, MDS 84\%.

Individual pt data can also be entered into the model, with a probability of whether the diagnosis is MDS vs. CMML provided along with the impact of each variable on the decision, as shown in Figure 1.

When the analysis was restricted to mutations only, the accuracy of the model dropped dramatically (77\%, ROC-AUC .85).


\subsection{Task 3}%

We propose a novel approach using interpretable, individualized modeling to predict MDS vs. CMML phenotypes based on genomic and clinical data without the need for bone marrow biopsy data. This approach can aid clinicians and hematopathologists when encountering pts with cytopenias and a diagnosis suspicious for MDS vs. CMML. The model also provides feature attributions that allow for quantitative understanding of the complex interplay among genotype, clinical variables, and phenotype.


